tests/demo_tests/test_ip_v4v6_raw_capture.py

```diff
@@ -0,0 +1,196 @@
+    eth.dst.decrement.start = dst
+    eth.dst.decrement.step = mac_step
+    eth.dst.decrement.count = 10
+    ipv6.src.increment.start = src_ipv6
+    ipv6.src.increment.step = ipv6_step
+    ipv6.src.increment.count = 10
+    ipv6.dst.increment.start = dst_ipv6
+    ipv6.dst.increment.step = ipv6_step
+    ipv6.dst.increment.count = 10
+    tcp.src_port.increment.start = "5000"
+    tcp.src_port.increment.step = "1"
+    tcp.src_port.increment.count = 10
+    tcp.dst_port.increment.start = "2000"
+    tcp.dst_port.increment.step = "1"
+    tcp.dst_port.increment.count = 10
+    f2.size.fixed = size
+    f2.duration.fixed_packets.packets = packets
+
+    ##############################
+    # Starting transmit on flows
+    ##############################
+    utils.start_traffic(app_intf, config)
+
+    ##############################
+    # Analyzing Traffic metrics
+    ##############################
+    print('Analyzing flow and port metrics ...')
+    utils.wait_for(
+        lambda: results_ok(app_intf, utils, size, packets * 2),
+        'stats to be as expected', timeout_seconds=10
+    )
+
+    ##############################
+    # Stopping transmit on flows
+    ##############################
+    print('Stopping transmit ...')
+    utils.stop_traffic(app_intf)
+
+    captures_ok(app_intf, config, utils, size, packets * 2)
+
+
+def results_ok(api, utils, size, packets):
+    """
+    Returns true if stats are as expected, false otherwise.
+    """
+    port_results, flow_results = utils.get_all_stats(api)
+    frames_ok = utils.total_frames_ok(port_results, flow_results, packets)
+    bytes_ok = utils.total_bytes_ok(
+        port_results, flow_results, packets * size
+    )
+    return frames_ok and bytes_ok
+
+
+def captures_ok(api, cfg, utils, size, packets):
+    """
+    Returns normally if patterns in captured packets are as expected.
+    """
+    src_mac = [[0x00, 0x10, 0x10, 0x20, 0x20, 0x10 + i] for i in range(10)]
+    dst_mac = [[0x00, 0x10, 0x10, 0x20, 0x20, 0x20 - i] for i in range(10)]
+    src_ip = [[0x0a, 0x01, 0x01 + i, 0x01] for i in range(10)]
+    dst_ip = [[0x0a, 0x01, 0x01 + i, 0x02] for i in range(10)]
+    src_ip6 = [
+        [
+            0xab, 0xcd + i, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x1a
+        ]
+        for i in range(10)
+    ]
+    dst_ip6 = [
+        [
+            0xab, 0xcd + i, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x2a
+        ]
+        for i in range(10)
+    ]
+    src_port = [[0x13, 0x88 + i] for i in range(10)]
+    dst_port = [[0x07, 0xd0 + i] for i in range(10)]
+
+    cap_dict = utils.get_all_captures(api, cfg)
+    assert len(cap_dict) == 1
+    size_dt = {
+        size: [0 for i in range(10)]
+    }
+
+    for index, b in enumerate(cap_dict[list(cap_dict.keys())[0]]):
+        try:
+            i = dst_mac.index(b[0:6])
+        except Exception:
```

Comment:
add a comment describing why exception might be raised here (due to unexpected packets)

Reply:
addressed